tracks/scala/exercises/accumulate/mentoring.md

```diff
@@ -0,0 +1,44 @@
+### Reasonable solutions
+
+#### Solution using immutable collection
+
+```scala
+import scala.annotation.tailrec
+
+class Accumulate {
+  def accumulate[A, B](f: (A) => B, list : List[A]): List[B] = accumulateHelper(f, list, List.empty)
+
+  @tailrec
+  private def accumulateHelper[A, B](f: (A) => B, list : List[A], result : List[B]): List[B] =
+    list match {
+      case Nil => result.reverse
+      case x :: xs => accumulateHelper(f, xs, f(x) :: result)
+    }
+}
+```
+
+#### Solution using mutable collection
```

Comment:
This is the same header as the previous solution. Perhaps use another header?

Reply:
I am confused. The first sample solution states 'immutable' collection. The second solution states ' mutable' collection? I don't understand what you are pointing out.

Reply:
I completely misread that, sorry about that! I read immutable twice.

Reply:
No worries. Thanks for reviewing.